projects/packages/my-jetpack/_inc/data/recommendations-section/use-recommendations-section.ts

```diff
@@ -25,9 +38,42 @@ const useRecommendationsSection = () => {
 		errorMessage: __( 'Failed to save the evaluation. Please try again', 'jetpack-my-jetpack' ),
 	} );
 
+	const submitEvaluation = useCallback(
+		( goals: string[] ) =>
+			new Promise< SubmitRecommendationsResult >( ( resolve, reject ) => {
+				handleSubmitRecommendations(
+					{ queryParams: { goals } },
+					{
+						onSuccess: resolve,
+						onError: reject,
+					}
+				);
+			} ),
+		[ handleSubmitRecommendations ]
+	);
+
+	const saveEvaluationResult = useCallback(
+		( recommendations: SubmitRecommendationsResult ) =>
+			new Promise< void >( ( resolve, reject ) => {
+				handleSaveEvaluationResult(
+					{ data: { recommendations } },
+					{
+						onSuccess: response => {
+							setRecommendedModules( response );
+							resolve();
+						},
+						onError: reject,
+					}
+				);
+			} ),
+		[ handleSaveEvaluationResult, setRecommendedModules ]
+	);
+
 	return {
 		submitEvaluation,
 		saveEvaluationResult,
+		recommendedModules,
+		isSectionVisible: null != recommendedModules && ! isWelcomeBannerVisible,
```

Comment:
Minor nitpick, if you are checking whether `recommendedModules` is exactly null, I'd recommend using a strict comparison instead. Also, yoda conditions are not required in JS (especially nowadays).

Reply:
> Question, how is it related to the version of JS? Now you got me curious 😄

I meant more in general, actually. It was very useful in the past when linters or other static analysis tools weren't that powerful and we had to "manually" prevent typos like `if ( $value = 42 )` when we actually wanted to compare (not assign) values. Since now we have editors, pre-commit hooks, CI, etc running static analysis tools, it's much easier to catch those errors.

Reply:
Thanks for the explanation!! That makes sense indeed.